### PR TITLE
Backport from 9.6

### PR DESCRIPTION
**Pull request by lucernae**
Add ARCHIVE_FILENAME parameters to specify fixed filename (#11)

* Add ARCHIVE_FILENAME parameters to specify fixed filename
* Update travis.yml for latest changes
* Edit README.md to reflect the changes
* Edit docker-compose.yml for latest update
* Add restore script

### CODE REVIEW

**Comment by lucernae** [comment]
@NyakudyaA @timlinux Do you have any comment on this?

File: restore.sh [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+echo "Dropping target DB"
+dropdb ${TARGET_DB}
+
+if [ -z "${WITH_POSTGIS:-}" ]; then
+	echo "Recreate target DB without POSTGIS"
+	createdb -O ${PGUSER} ${TARGET_DB}
+else
+	echo "Recreate target DB with POSTGIS"
+	createdb -O ${PGUSER} -T template_postgis ${TARGET_DB}
```
Comment (NyakudyaA) [inline restore.sh]: @lucernae we are trying to do away with the template method for installing PostGIS see https://github.com/kartoza/docker-postgis/pull/119

**Comment by NyakudyaA** [comment]
@lucernae The backup script does not backup globals (users and permissions).  We should include it `pg_dumpall -h localhost -p 5432 -U docker -v --globals-only -f "/path/to/globals.sql"`

This is quite important as the starting point for any restore should always be
* Create a DB
* Restore users and permissions
* Restore the database

**Comment by lucernae** [comment]
@NyakudyaA I see. There are some parts that needs to be changed.

> @lucernae we are trying to do away with the template method for installing PostGIS see [kartoza/docker-postgis#119](https://github.com/kartoza/docker-postgis/pull/119)

Yes! it will help the restore process if we can get what extensions to include.

> @lucernae The backup script does not backup globals (users and permissions). We should include it `pg_dumpall -h localhost -p 5432 -U docker -v --globals-only -f "/path/to/globals.sql"`

We might have to change the backup script for this.

Can I request that both changes be done in a separate issue/PR after this, since those changes is specific for the latest kartoza/postgis base images.

I'm trying to keep this PR for backport changes only from 9.6 to master.
Which includes:

* options to create backup archive by specified filename only (this is used to chain with borg deduplication)
* rewrite travis tests
* add docs

Do we have any more feedback for those points?

**Comment by NyakudyaA** [comment]
@lucernae LGTM we can open another issue for those things I mentioned

**Pull request merged**